src/main/java/org/opensearch/neuralsearch/processor/rerank/ByFieldRerankProcessor.java

```diff
@@ -0,0 +1,159 @@
+     * @param targetField           The field you want to replace your <code>_score</code> with
+     * @param removeTargetField     A flag to let you delete the target_field for better visualization (i.e. removes a duplicate value)
+     * @param keepPreviousScore     A flag to let you decide to stash your previous <code>_score</code> in a field called <code>previous_score</code> (i.e. for debugging purposes)
+     * @param contextSourceFetchers  Context from some source and puts it in a map for a reranking processor to use <b> (Unused in ByFieldRerankProcessor)</b>
+     */
+    public ByFieldRerankProcessor(
+        String description,
+        String tag,
+        boolean ignoreFailure,
+        String targetField,
+        boolean removeTargetField,
+        boolean keepPreviousScore,
+        final List<ContextSourceFetcher> contextSourceFetchers
+    ) {
+        super(RerankType.BY_FIELD, description, tag, ignoreFailure, contextSourceFetchers);
+        this.targetField = targetField;
+        this.removeTargetField = removeTargetField;
+        this.keepPreviousScore = keepPreviousScore;
+    }
+
+    @Override
+    public void rescoreSearchResponse(SearchResponse response, Map<String, Object> rerankingContext, ActionListener<List<Float>> listener) {
+        SearchHit[] searchHits = response.getHits().getHits();
+
+        SearchHitValidator searchHitValidator = this::byFieldSearchHitValidator;
+
+        if (!validateRerankCriteria(searchHits, searchHitValidator, listener)) {
+            return;
+        }
+
+        List<Float> scores = new ArrayList<>(searchHits.length);
+
+        for (SearchHit hit : searchHits) {
+            Map<String, Object> sourceAsMap = hit.getSourceAsMap();
+
+            float score = getScoreFromSourceMap(sourceAsMap, targetField);
+            scores.add(score);
+
+            if (keepPreviousScore) {
+                sourceAsMap.put("previous_score", hit.getScore());
+            }
+
+            if (removeTargetField) {
+                removeTargetFieldFromSource(sourceAsMap, targetField);
+            }
+
+            try {
+                XContentBuilder builder = XContentBuilder.builder(XContentType.JSON.xContent());
+                BytesReference sourceMapAsBytes = BytesReference.bytes(builder.map(sourceAsMap));
+                hit.sourceRef(sourceMapAsBytes);
+            } catch (IOException e) {
+                listener.onFailure(new RuntimeException(e));
+                return;
+            }
+        }
+
+        listener.onResponse(scores);
+    }
+
+    public void byFieldSearchHitValidator(SearchHit hit) {
+        if (!hit.hasSource()) {
+            throw new IllegalArgumentException("There is no source field to be able to perform rerank on hit [" + hit.docId() + "]");
+        }
+
+        Map<String, Object> sourceMap = hit.getSourceAsMap();
+        if (!mappingExistsInSource(sourceMap, targetField)) {
```

Comment:
when one of the hit missing the field, it will throw exception. Have you test when one of the field is missing and ignore Failure, what is the response?

Reply:
Hey, thanks for the test case I just ran it and it still threw the exception. I believe this is because we are working with in the search response level If this were in a end to end workflow within search pipelines then the original search response would be returned as it states in [SearchPipeline](https://github.com/opensearch-project/OpenSearch/blob/c20458512ff4e68f359de32bfad91a6a7487f8a1/server/src/main/java/org/opensearch/search/pipeline/Pipeline.java#L238) class